### PR TITLE
Move subsetCheck to Set Interface

### PR DESCRIPTION
**Pull request by shibumi**
**Fixes issue #**:
https://github.com/in-toto/in-toto-golang/issues/66

**Description of pull request**:
This Requests fulfils issue #66 and moves the subsetCheck function into our Set interface.

**Please verify and check that the pull request fulfills the following
requirements**:

- [X] Tests have been added for the bug fix or new feature
- [X] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by shibumi** [comment]
This PR is ready to merge

**Comment by shibumi** [comment]
@lukpueh you might want to have another look :)

File: in_toto/util_test.go [hunk 1]
```diff
@@ -96,15 +96,15 @@ func TestInterfaceKeyStrings(t *testing.T) {
...
-		superset []string
+		superset Set
 		result   bool
 	}{
-		{[]string{"sha256"}, []string{"sha256", "sha512"}, true},
-		{[]string{"sha512"}, []string{"sha256"}, false},
-		{[]string{"sha256", "sha512"}, []string{"sha128", "sha256", "sha512"}, true},
+		{[]string{"sha256"}, Set{"sha256": {}, "sha512": {}}, true},
+		{[]string{"sha512"}, Set{"sha256": {}}, false},
+		{[]string{"sha256", "sha512"}, Set{"sha128": {}, "sha256": {}, "sha512": {}}, true},
```
Comment (lukpueh) [inline in_toto/util_test.go:104]: Minor nit: I think it's a tad more readable if you use e.g. `NewSet{"sha256",  "sha512"}`, etc.

File: in_toto/util.go [hunk 2]
```diff
@@ -132,25 +132,15 @@ func InterfaceKeyStrings(m map[string]interface{}) []string {
...
-	// TODO: This function might be better as addition
-	// to our Set interface.
-	// We use a Go label here to break out to the outer loop
-OUTER:
-	for _, sub := range subset {
-		for _, super := range superset {
-			if sub == super {
-				continue OUTER
-			}
+func (s Set) IsSubSet(subset Set) bool {
```
Comment (lukpueh) [inline in_toto/util.go:137]: Minor performance nit: You could do a quick len check to return early (subset can't be greater than superset).

File: in_toto/util.go [hunk 3]
```diff
@@ -132,25 +132,15 @@ func InterfaceKeyStrings(m map[string]interface{}) []string {
 
 /*
 subsetCheck checks if all strings in a slice of strings
-can be found in a superset slice of strings.
+can be found in a Set.
```
Comment (lukpueh) [inline in_toto/util.go:135]: Please update the docstring, also so that it is clear which one is expected to be the subset and which the superset.

**Review by lukpueh — CHANGES_REQUESTED** [review]
Thanks for the changes! Almost there. :)

**Comment by shibumi** [comment]
@lukpueh all done.

**Review by lukpueh — APPROVED** [review]
💯

**Pull request merged**